src/main/java/net/ripe/ipresource/ImmutableResourceSet.java

```diff
@@ -65,6 +65,9 @@ private ImmutableResourceSet() {
     }
 
     private ImmutableResourceSet(TreeMap<UniqueIpResource, IpResource> resourcesByEndPoint) {
+        if (resourcesByEndPoint.isEmpty()) {
```

Comment:
Is that a convenient API here? Can we return the empty instance instead?

Reply:
This cannot be done from the constructor, which is why it is private and you have to use one of the factory `of` or `empty` methods. So this check is more of an internal assertion than something that can be triggered by the user of this class.